File: data/plugins/skill/mining/pickaxe.rb
require 'java'

java_import 'org.apollo.game.model.Animation'

PICKAXES = {}
PICKAXE_IDS = []

# A pickaxe that can be mined with.
class Pickaxe
  attr_reader :id, :level, :animation, :pulses, :ratio

  def initialize(id, level, animation, pulses, ratio)
    @id = id
    @level = level
    @animation = Animation.new(animation)
    @pulses = pulses
    @ratio = ratio
  end
end

def append_pickaxe(pickaxe)
  PICKAXES[pickaxe.id] = pickaxe
  PICKAXE_IDS << pickaxe.id
end

# NOTE: ADD LOWER LEVEL PICKAXES FIRST
append_pickaxe(Pickaxe.new(1265, 1,  625, 8, 0.05)) # bronze pickaxe
append_pickaxe(Pickaxe.new(1267, 1,  626, 7, 0.1)) # iron pickaxe
append_pickaxe(Pickaxe.new(1269, 6,  627, 6, 0.2)) # steel pickaxe
append_pickaxe(Pickaxe.new(1273, 21, 629, 5, 0.3)) # mithril pickaxe
append_pickaxe(Pickaxe.new(1271, 31, 628, 4, 0.45)) # adamant pickaxe
append_pickaxe(Pickaxe.new(1275, 41, 624, 3, 0.65)) # rune pickaxe

PICKAXE_IDS.reverse!

File: game/plugin/skills/mining/src/pickaxe.kt
package org.apollo.game.plugin.skills.mining

import org.apollo.game.model.Animation;

enum class Pickaxe(val id: Int, val level: Int, val animation: Animation, val pulses: Int) {
    RUNE(1275, 41, Animation(624), 3),
    ADAMANT(1271, 31, Animation(628), 4),
    MITHRIL(1273, 21, Animation(629), 5),
    STEEL(1269, 1, Animation(627), 6),
    ITRON(1267, 1, Animation(626), 7),
    BRONZE(1265, 1, Animation(625), 8)
}

val PICKAXES = Pickaxe.values()



fun getPickaxes(): Array<Pickaxe> {
    return PICKAXES
}

fun lookupPickaxe(id: Int): Pickaxe? = PICKAXES.find { it.id == id }

